data/store/mongo/mongo.go

```diff
@@ -635,9 +635,9 @@ func (d *DataRepository) UnarchiveDeviceDataUsingHashesFromDataSet(ctx context.C
 
 	result := struct {
 		ID struct {
-			Active            bool   `bson:"_active"`
-			ArchivedDataSetID string `bson:"archivedDatasetId"`
-			ArchivedTime      string `bson:"archivedTime"`
+			Active            bool       `bson:"_active"`
+			ArchivedDataSetID *string    `bson:"archivedDatasetId"`
+			ArchivedTime      *time.Time `bson:"archivedTime"`
```

Comment:
Now that I read the code again it might not even be possible to have the record in this state. This is the code before the changes:
```golang
ArchivedTime string `bson:"archivedTime"`

...

if result.ID.Active != (result.ID.ArchivedDataSetID == "") || result.ID.Active != (result.ID.ArchivedTime == "") {
   continue
}

if result.ID.Active {
...
} else {
	set["archivedDatasetId"] = result.ID.ArchivedDataSetID
	set["archivedTime"] = result.ID.ArchivedTime
}
```

Reply:
I want to read the code history on this a bit more, but at this point it seems like it is an impossible state, and hopefully this never becomes a problem.

Though, as part of the migration, old records should be checked for empty dates, in the case that there was a previously acceptable state where empty dates were possible.

Reply:
History shows no changes in logic that should affect this, it should™ be fine, but still worth checking during migration.